frontend/containers/UserProfileEdit/UserProfileEdit.jsx

```diff
@@ -3,15 +3,49 @@
 import {h, Component} from 'preact'
 import {connect} from 'preact-redux'
 import {bindActionCreators} from 'redux'
+import {route} from 'preact-router'
 
-import {connectHelper} from 'lib/util'
+import {connectHelper, setPageTitle, Case} from 'lib/util'
+import {getAuthCollection} from 'lib/app-config'
+import {buildUrl} from 'lib/router'
 
-import * as documentActions from 'actions/documentActions'
+import * as userActions from 'actions/userActions'
 
 /**
  * The interface for editing a user profile.
  */
 class UserProfileEdit extends Component {
+
+  componentDidUpdate(previousProps) {
+    const {
+      actions,
+      section,
+      state
+    } = this.props
+
+    const sections = ['account', 'settings', 'password_reset']
+
+    if (state.app.config) {
+      const auth = state.app.config.auth
+
+      const currentCollection = getAuthCollection(state.api.apis, auth)
+
+      if (currentCollection) {
+        const fields = Object.keys(currentCollection.fields)
+        const firstField = Object.keys(currentCollection.fields).find(field => (currentCollection.fields[field].publish && currentCollection.fields[field].publish.section))
```

Comment:
Didn't you mean to use `fields` in `firstField`, instead of repeating `Object.keys`?